crates/tabby-scheduler/src/lib.rs

```diff
@@ -15,8 +15,12 @@ use tracing::{error, info, warn};
 pub async fn scheduler<T: RepositoryAccess + 'static>(now: bool, access: T) -> Result<()> {
     if now {
         let repositories = access.list_repositories().await?;
-        job_sync(&repositories);
-        job_index(&repositories);
+        if !job_sync(&repositories) {
+            panic!("Sync job failed, exiting");
```

Comment:
don't panic - propagate the error so it can be handled in main:

https://github.com/TabbyML/tabby/blob/f94b5978e63930adf2ca1f636d2e6f01bf3ed60a/crates/tabby/src/main.rs#L159